pymeasure/instruments/agilent/agilentL4534A.py

```diff
@@ -0,0 +1,384 @@
+    if 'filter' not in value:
+        raise ValueError('Requires \'filter\'')
+
+    strict_discrete_set(value['range'], VOLTAGE_RANGE_VALUES)
+    strict_discrete_set(value['coupling'], COUPLING_VALUES)
+    strict_discrete_set(value['filter'], FILTER_VALUES)
+    return value
+
+
+class AgilentL4534A(Instrument):
+    """
+    Represents the Agilent L4532A/L4534A digitizers.
+    """
+    def __init__(self, adapter, name="Agilent L4534A Digitizer", **kwargs):
+        super().__init__(
+            adapter, name, **kwargs
+        )
+
+    SAMPLE_RATE_VALUES = SAMPLE_RATE_VALUES
+
+    class DigitizerChannel(Channel):
+        """Implementation of an Agilent L4532A/L4534A channel."""
+
+        def __init__(self, instrument, id):
+            super().__init__(instrument, id)
+
+        VOLTAGE_RANGE_VALUES = VOLTAGE_RANGE_VALUES
+        COUPLING_VALUES = COUPLING_VALUES
+        FILTER_VALUES = FILTER_VALUES
+
+        config = Instrument.control(
+            "CONF:CHAN:ATTR? (@{ch})",
+            "CONF:CHAN:ATTR (@{ch}),%s",
+            """
+            Control Channel configuration with dict containing range (in V), coupling, and filter.
+            """,
+            set_process=lambda v: '{:.3g},{},{}'.format(
+                assume_units(v['range'], ureg.V).m_as(ureg.V), v['coupling'], v['filter']),
+            get_process=_get_channel_config_process,
+            validator=_validate_channel_config
+        )
+
+        range = Instrument.control(
+            "CONF:CHAN:RANG? (@{ch})",
+            "CONF:CHAN:RANG (@{ch}),%s",
+            """
+            Control Voltage range for this channel (0.25, 0.5, 1, 2, 4, 8, 16, 32, 128, 256).
+            """,
+            set_process=lambda v: Decimal(assume_units(v, ureg.V).m_as(ureg.V)).to_eng_string(
+            ),  # send the value as V to the device
+            get_process=lambda v: ureg.Quantity(v, ureg.V),  # convert to quantity
+            validator=strict_discrete_set,
+            values=VOLTAGE_RANGE_VALUES
+        )
+
+        coupling = Instrument.control(
+            "CONF:CHAN:COUP? (@{ch})",
+            "CONF:CHAN:COUP (@{ch}),%s",
+            """Comtrol channel coupling (AC|DC).""",
+            validator=strict_discrete_set,
+            values=COUPLING_VALUES
+        )
+
+        filter = Instrument.control(
+            "CONF:CHAN:FILT? (@{ch})",
+            "CONF:CHAN:FILT (@{ch}),%s",
+            """Control Filter for this channel (LP_20_MHZ, LP_2_MHZ, LP_200_KHZ).""",
+            validator=strict_discrete_set,
+            values=FILTER_VALUES
+        )
+
+        def _read_data_block(self):
+            # Data block has header of form '#<n><n digits>',
+            # where <n>specifies the number of digits used to indicate the size of the block,
+            # and the <n digits> indicate the block size in bytes.
+            digits = self.read_bytes(2).decode()
+            if not digits.startswith('#'):
+                raise Error()
```

Comment:
@bmoneke This is the only place `Error` is used in the codebase, besides being passed directly to [`RangeError`](https://github.com/pymeasure/pymeasure/blob/3c288c7909acc5cd3c36af425f41b1e891152bae/pymeasure/errors.py)

I think `pymeasure/errors.py` would be for defining pymeasure-specific error types, but a generic `Error` type seems too broad. Should `Error` be modified to `InstrumentError` or something more pymeasure-y or be replaced with a builtin `Exception` type?

Also I think there should be a user-friendly message that is passed with this statement if `digits` doesn't start with `#`.